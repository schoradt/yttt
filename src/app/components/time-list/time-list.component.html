<div class="time-list">
  <mat-card appearance="outlined">
    <mat-card-header>
      <mat-card-title>Track Time</mat-card-title>
      <mat-card-subtitle>New entry</mat-card-subtitle>
    </mat-card-header>
    <mat-card-content>
      <table mat-table [dataSource]="timeList" class="mat-elevation-z8">

        <!-- Position Column -->
        <ng-container matColumnDef="start">
          <th mat-header-cell *matHeaderCellDef> Start </th>
          <td mat-cell *matCellDef="let element"> {{element.startTime}} </td>
        </ng-container>

        <!-- Name Column -->
        <ng-container matColumnDef="end">
          <th mat-header-cell *matHeaderCellDef> End </th>
          <td mat-cell *matCellDef="let element"> {{element.endTime}} </td>
        </ng-container>

        <!-- Weight Column -->
        <ng-container matColumnDef="ticket">
          <th mat-header-cell *matHeaderCellDef> Ticket </th>
          <td mat-cell *matCellDef="let element"> {{element.ticket}} </td>
        </ng-container>

        <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
        <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
      </table>
    </mat-card-content>
  </mat-card>
</div>
