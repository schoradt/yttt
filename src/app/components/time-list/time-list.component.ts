import { Component, Input } from '@angular/core';
import { MatTableModule } from '@angular/material/table';
import { MatCardModule } from '@angular/material/card';
import { TimeTrack } from '../../model/time-track';

@Component({
  selector: 'yttt-time-list',
  imports: [MatTableModule, MatCardModule],
  templateUrl: './time-list.component.html',
  styleUrl: './time-list.component.scss'
})
export class TimeListComponent {
  @Input()
  timeList: TimeTrack[] = [];

  displayedColumns: string[] = ['start', 'end', 'ticket'];
}
