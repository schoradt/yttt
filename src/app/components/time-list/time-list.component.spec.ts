import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TimeListComponent } from './time-list.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe('TimeListComponent', () => {
  let component: TimeListComponent;
  let fixture: ComponentFixture<TimeListComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [TimeListComponent, NoopAnimationsModule]
    }).compileComponents();

    fixture = TestBed.createComponent(TimeListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should be rendered empty', () => {
    component.timeList = [];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should be rendered with one element', () => {
    component.timeList = [
      {
        ticket: 'T1',
        startTime: new Date('2025-01-12T10:00:00Z')
      }
    ];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
