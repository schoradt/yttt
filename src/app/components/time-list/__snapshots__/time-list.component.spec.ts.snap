// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeListComponent should be rendered empty 1`] = `
<yttt-time-list
  displayedColumns={[Function Array]}
  timeList={[Function Array]}
>
  <div
    class="time-list"
  >
    <mat-card
      appearance="outlined"
      class="mat-mdc-card mdc-card mat-mdc-card-outlined mdc-card--outlined"
    >
      <mat-card-header
        class="mat-mdc-card-header"
      >
        <div
          class="mat-mdc-card-header-text"
        >
          <mat-card-title
            class="mat-mdc-card-title"
          >
            Track Time
          </mat-card-title>
          <mat-card-subtitle
            class="mat-mdc-card-subtitle"
          >
            New entry
          </mat-card-subtitle>
        </div>
      </mat-card-header>
      <mat-card-content
        class="mat-mdc-card-content"
      >
        <table
          class="mat-mdc-table mdc-data-table__table cdk-table mat-elevation-z8"
          mat-table=""
          role="table"
        >
          <thead
            role="rowgroup"
          >
            <tr
              class="mat-mdc-header-row mdc-data-table__header-row cdk-header-row"
              mat-header-row=""
              role="row"
            >
              <th
                class="mat-mdc-header-cell mdc-data-table__header-cell cdk-header-cell cdk-column-start mat-column-start"
                mat-header-cell=""
                role="columnheader"
              >
                 Start 
              </th>
              <th
                class="mat-mdc-header-cell mdc-data-table__header-cell cdk-header-cell cdk-column-end mat-column-end"
                mat-header-cell=""
                role="columnheader"
              >
                 End 
              </th>
              <th
                class="mat-mdc-header-cell mdc-data-table__header-cell cdk-header-cell cdk-column-ticket mat-column-ticket"
                mat-header-cell=""
                role="columnheader"
              >
                 Ticket 
              </th>
            </tr>
          </thead>
          <tbody
            class="mdc-data-table__content"
            role="rowgroup"
          >
          </tbody>
          <tfoot
            class="mat-mdc-table-sticky"
            role="rowgroup"
            style="display: none; bottom: 0px; z-index: 10;"
          >
          </tfoot>
        </table>
      </mat-card-content>
    </mat-card>
  </div>
</yttt-time-list>
`;

exports[`TimeListComponent should be rendered with one element 1`] = `
<yttt-time-list
  displayedColumns={[Function Array]}
  timeList={[Function Array]}
>
  <div
    class="time-list"
  >
    <mat-card
      appearance="outlined"
      class="mat-mdc-card mdc-card mat-mdc-card-outlined mdc-card--outlined"
    >
      <mat-card-header
        class="mat-mdc-card-header"
      >
        <div
          class="mat-mdc-card-header-text"
        >
          <mat-card-title
            class="mat-mdc-card-title"
          >
            Track Time
          </mat-card-title>
          <mat-card-subtitle
            class="mat-mdc-card-subtitle"
          >
            New entry
          </mat-card-subtitle>
        </div>
      </mat-card-header>
      <mat-card-content
        class="mat-mdc-card-content"
      >
        <table
          class="mat-mdc-table mdc-data-table__table cdk-table mat-elevation-z8"
          mat-table=""
          role="table"
        >
          <thead
            role="rowgroup"
          >
            <tr
              class="mat-mdc-header-row mdc-data-table__header-row cdk-header-row"
              mat-header-row=""
              role="row"
            >
              <th
                class="mat-mdc-header-cell mdc-data-table__header-cell cdk-header-cell cdk-column-start mat-column-start"
                mat-header-cell=""
                role="columnheader"
              >
                 Start 
              </th>
              <th
                class="mat-mdc-header-cell mdc-data-table__header-cell cdk-header-cell cdk-column-end mat-column-end"
                mat-header-cell=""
                role="columnheader"
              >
                 End 
              </th>
              <th
                class="mat-mdc-header-cell mdc-data-table__header-cell cdk-header-cell cdk-column-ticket mat-column-ticket"
                mat-header-cell=""
                role="columnheader"
              >
                 Ticket 
              </th>
            </tr>
          </thead>
          <tbody
            class="mdc-data-table__content"
            role="rowgroup"
          >
            <tr
              class="mat-mdc-row mdc-data-table__row cdk-row"
              mat-row=""
              role="row"
            >
              <td
                class="mat-mdc-cell mdc-data-table__cell cdk-cell cdk-column-start mat-column-start"
                mat-cell=""
                role="cell"
              >
                 Sun Jan 12 2025 10:00:00 GMT+0000 (Greenwich Mean Time) 
              </td>
              <td
                class="mat-mdc-cell mdc-data-table__cell cdk-cell cdk-column-end mat-column-end"
                mat-cell=""
                role="cell"
              >
              </td>
              <td
                class="mat-mdc-cell mdc-data-table__cell cdk-cell cdk-column-ticket mat-column-ticket"
                mat-cell=""
                role="cell"
              >
                 T1 
              </td>
            </tr>
          </tbody>
          <tfoot
            class="mat-mdc-table-sticky"
            role="rowgroup"
            style="display: none; bottom: 0px; z-index: 10;"
          >
          </tfoot>
        </table>
      </mat-card-content>
    </mat-card>
  </div>
</yttt-time-list>
`;
